src/sqlfluff/utils/reflow/elements.py

```diff
@@ -28,11 +28,26 @@ class ReflowElement:
     def _class_types(segments: Sequence[RawSegment]) -> Set[str]:
         return set(chain.from_iterable(seg.class_types for seg in segments))
 
+    @property
+    def class_types(self):
+        """The set of contained class types.
+
+        Parallel to BaseSegment.class_types
+        """
+        return self._class_types(self.segments)
+
     @property
     def raw(self):
         """Get the current raw representation."""
         return "".join(seg.raw for seg in self.segments)
 
+    def num_newlines(self) -> int:
+        """How many newlines does this element contain?"""
+        return sum(
+            bool(seg.class_types.intersection({"newline", "end_of_file"}))
```

Comment:
Any concern with equating end of file with newline? Especially if (as is typical) the last character in a file is a newline.

Reply:
This is a really good point. I've actually changed the handling of this in the next PR, but I think there's an argument for me to push that down into this one. I'll do that so that it's handled consistently from the start.